web-console/src/utils/sampler.ts

```diff
@@ -63,6 +63,38 @@ export interface SamplerConfig {
 
 export interface SampleResponse {
   data: SampleEntry[];
+  logicalSegmentSchema: { name: string; type: string }[];
+  logicalDimensions: DimensionSpec[];
+  physicalDimensions: DimensionSpec[];
+  numRowsIndexed: number;
+  numRowsRead: number;
+}
+
+export function getHeaderNamesFromSampleResponse(
+  sampleResponse: SampleResponse,
+  ignoreTimeColumn = false,
+) {
+  return filterMap(sampleResponse.logicalSegmentSchema, s =>
+    ignoreTimeColumn && s.name === '__time' ? undefined : s.name,
+  );
+}
+
+export function guessDimensionsFromSampleResponse(sampleResponse: SampleResponse): DimensionSpec[] {
+  const { logicalDimensions, physicalDimensions, data } = sampleResponse;
+  return logicalDimensions.map(d => {
+    // Boolean column are currently reported as "long" so let's turn them into "string"
```

Comment:
im still looking into this, since technically looking at stuff it looks like it should be dependent on the value of `druid.expressions.useStrictBooleans`. Additionally, 'long' really probably is better when using 'auto', since longs do have indexes in this mode, so im a bit conflicted about this staying like this long term, but i think its fine at least until we make 'auto' the default schemaless (or add indexes to classic 'long' schema)

Reply:
the reason it is picking string is that if you pick long then it forces the `true` / `false` in the input to `null` (instead of `1` / `0`)